File: Models/SpaceObject.cs
namespace TestSystem.Models
{
    public class SpaceObject
    {
        private string? _type;
        private string? _name;
        private ulong _age = 0;
        private uint _diameter = 0;
        private uint _weight = 0;

        SpaceObject() { }
        SpaceObject(string type, string name, ulong age, uint diametr, uint weight)
        {
            Type = type;
            Name = name;
            Age = age;
            Diameter = diametr;
            Weight = weight;
        }

        public int Id { get; set; }
        public List<SpaceSystem> SpaceSystem = new();
        public string? Type 
        { 
            get { return _type; }
            set { if (value != null) _type = value.Trim().ToLower(); }
        }
        public string? Name
        {
            get =>  _name;
            
            set { if (value != null) _name = value; }
        }
        public ulong Age
        {
            get => _age;
            set { _age = value; }
        }

        public uint Diameter
        {
            get => _diameter; 
            set { _diameter = value; }
        }
        public uint Weight
        {
            get => _weight;
            set { _weight = value; }
        }
    }
}

File: client/src/SpaceObject.ts
import {SpaceSystem} from './SpaceSystem'

export class SpaceObject
{
        private _type:  string = '';
        private _name: string = '';
        private  _age = 0;
        private _diameter = 0;
        private _weight = 0;
        public Id!: number;
        public spaceSystem!: SpaceSystem[];

    constructor(type: string, name: string, age: number, diameter: number, weight: number){
        this._type = type;
        this._age = age;
        this._diameter = diameter;
        this._weight = weight;
    }

        public get type():string
        {
            return this._type;
        }
        public set type(s: string)
        {
            this._type = s;
        }

        public get name():string
        {
            return this._name;
        }
        public set name(s: string)
        {
            this._name = s;
        }

        public get age():number
        {
            return this._age;
        }
        public set age(a: number)
        {
            this._age = a;
        }

        public get diameter():number
        {
            return this._diameter;
        }
        public set diameter(a: number)
        {
            this._diameter = a;
        }

        public get weight():number
        {
            return this._weight;
        }
        public set weight(a: number)
        {
            this._weight = a;
        }
}